integrations/google/gemini/client.go

```diff
@@ -31,6 +37,58 @@ func New(cvars sdkservices.Vars) sdkservices.Integration {
 	return sdkintegrations.NewIntegration(
 		desc,
 		sdkmodule.New( /* No exported functions for Starlark */ ),
+		connTest(cvars),
 		sdkintegrations.WithConnectionConfigFromVars(cvars),
 	)
 }
+
+// connTest is an optional connection test provided by the integration
+// to AutoKitteh. It is used to verify that the connection is working
+// as expected. The possible results are "OK" and "error".
+func connTest(cvars sdkservices.Vars) sdkintegrations.OptFn {
+	return sdkintegrations.WithConnectionTest(func(ctx context.Context, cid sdktypes.ConnectionID) (sdktypes.Status, error) {
+		if !cid.IsValid() {
+			return sdktypes.NewStatus(sdktypes.StatusCodeWarning, "Init required"), nil
+		}
+
+		vs, err := cvars.Get(ctx, sdktypes.NewVarScopeID(cid))
+		if err != nil {
+			return sdktypes.InvalidStatus, err
+		}
+
+		apiKey := vs.Get(apiKeyVar)
+		if !apiKey.IsValid() || apiKey.Value() == "" {
+			return sdktypes.NewStatus(sdktypes.StatusCodeWarning, "Init required"), nil
+		}
+		url := "https://generativelanguage.googleapis.com/v1beta/models/gemini-1.5-flash:generateContent?key=" + apiKey.Value()
+		requestBody := `{
+				"contents": [{
+					"parts": [{"text": "Hello, Gemini!"}]
+				}]
+			}`
```

Comment:
I know I said not to add new Go libraries when HTTP is enough, but in this specific case (the API is still technically in beta) if might be safer to rely on Google's Go library instead of this hardcoded URL.

Also, I think a "cheaper"/quicker check might be this: https://pkg.go.dev/github.com/google/generative-ai-go/genai#Client.ListModels (like in ChatGPT)

Reply:
I think using the library simplifies the code significantly, but just as a side note:
> NOTE: This client uses the v1beta version of the API.

https://pkg.go.dev/github.com/google/generative-ai-go/genai#section-documentation